src/stratis_cli/_errors.py

```diff
@@ -30,6 +32,87 @@ class StratisCliRuntimeError(StratisCliError):
     """
 
 
+class StratisNoChangeError(StratisCliError):
+    """
+    Raised if there was a failure due to no changed state in stratisd's engine.
+    """
+
+    def __init__(self, command, resources):
+        """ Initializer.
+
+            :param str command: the executed command
+            :param str or list of str resources: the target resources
+        """
+        # pylint: disable=super-init-not-called
+        self.command = command
+        self.resources = resources
+
+    def __str__(self):
+        return "The command '{}' has already been completed for resource '{}'".format(
+            self.command, self.resources
+        )
+
+
+class StratisPartialChangeError(StratisCliError):
+    """
+    Raised if there was a failure due to partially changed state in stratisd's engine.
+    """
+
+    def __init__(self, command, changed_resources, unchanged_resources):
+        """ Initializer.
+
+            :param str command: the command run that caused the error
+            :param list of str changed_resources: the target resources that would change
+            :param list of str unchanged_resources: the target resources that would not change
+        """
+        # pylint: disable=super-init-not-called
+        self.command = command
+        self.changed_resources = changed_resources
+        self.unchanged_resources = unchanged_resources
+
+    def __str__(self):
+        msg = "The command '{}' has already been completed for resources {} ".format(
+            self.command, self.unchanged_resources
+        )
+        if self.changed_resources != []:
+            msg += "but not yet for resources {}".format(self.changed_resources)
+        return msg
+
+
+class StratisInUseError(StratisCliError):
+    """
+    Raised if a block device is added as both a blockdev and a cachedev
+    """
+
+    def __init__(self, blockdevs, added_as="cache"):
+        """ Initializer.
+
+            :param list of str blockdevs: the blockdevs that would be added in both tiers
+            :param str added_as: whether the action causing the error on the cache or data tier
```

Comment:
Please update the docs for ```added_as``` to correspond w/ the recent changes. Also, please make ```added_as``` a mandatory argument. The code for ```__str__``` will be shorter, and since this exception isn't part of any public API, we don't need to worry about future extensibility.

Reply:
Resolved.